content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+            m = nn.Softmax(dim=1)
+            outputs = m(outputs)
+            targets = targets.float().resize_(len(targets), 1)
+
+        return {"test_loss": loss}
+```
+
+Notice that in all functions we’ve defined as part of MyMEDMnistTrial, we have removed all instances of manual device management, metric logging, and accuracy calculations. HPE Machine Learning Development Environment’s Trial APIs handle these automatically, given only the Trial definition and configuration file.  
+
+After porting the model, compare the original training script and our newly defined model_def.py training file. At 302 vs. 177 lines of code, we have cut our training script nearly in half! 
+
+#### *Optional*: Upload dataset to S3 bucket  
+
+The PathMNIST dataset includes images that are preprocessed to 28x28 images with corresponding classification labels, complete with data augmentation. The original training script downloads this dataset and performs data normalization prior to model training.  
+
+When submitting experiments to HPE Machine Learning Development Environment, the dataset will be downloaded to the master in the same way, which may cause a small delay. To avoid this, upload your data to an S3 bucket and access it via the following instead: 
+
+Modify `config.yaml` to include the data url as follows: 
+
+```yaml
+data:
+  url: https://medmnist-pathmnist.s3.us-east-2.amazonaws.com/pathmnist.npz
+```
+
+Access the dataset via the following in the class initialization function: 
+
+```yaml
+os.makedirs(DATASET_ROOT, exist_ok=TRUE)
+wget.download(
+    context.get_data_config()["url"],
+    out=os.path.join(DATASET_ROOT, "pathmnist.npz"),
+)
+```
+
+#### Step 1.5 Train the Model 
+
+In HPE Machine Learning Development Environment, an experiment is a training job that consists of one or more variations, or trials, on the same model 
+
+To begin training our PathMNIST model, you need to submit an experiment to HPE Machine Learning Development Environment. After setting up Determined, use the CLI to submit an experiment (refer to full setup instructions in the [repository](https://github.com/ighodgao/determined_medmnist_e2e)): 
+
+`det e create config.yaml .` 
+
+View progress on the WebUI located at <DET_MASTER>:8080, which you can reach by pasting this address into a browser window. 
+
+You can configure additional training parameters by modifying config.yaml. Refer to the [Experiment Configuration Reference](https://docs.determined.ai/latest/reference/reference-training/experiment-config-reference.html#experiment-configuration-reference) for more information. 
+
+And that’s it! Now that we have our trained PathMNIST model, we can progress to deployment in Step 2.  
+
+#### Step 2: Model Deployment 
+
+Once you have finished training with HPE Machine Learning Development Environment, you can deploy your model using any solution.  
+
+In Step 2, we’ll show an example of using Flask to deploy your model as a RESTful API.  
```

Comment:
Modify to "In Step 2, you'll see an example of using Flask to deploy your model as a RESTful API."